torch/nested/_internal/nested_tensor.py

```diff
@@ -73,17 +76,27 @@ def __new__(
         )
         return r
 
-    def __init__(self, values, offsets, *, lengths=None, **kwargs):
+    def __init__(self, values, offsets, *, lengths=None, is_dummy=False, **kwargs):
         super().__init__()
         # Only support jagged for now.
         assert offsets is not None
         assert offsets.ndim == 1
         assert not isinstance(values, NestedTensor)
 
+        self._values = values
+        self._offsets = offsets
+        self._lengths = lengths
+
+        if is_dummy:
+            # Avoid infinite recursion
+            # Initialize the _values, _offsets, etc because they are checked
+            # when dispatching to jagged funcs in torch dispatch.
+            return
```

Comment:
Are you just trying to avoid the dummy NT having a singleton associated with its size? is this just to keep the mapping between singleton <-> offsets / lengths cleaner or is there an actual problem with it having one?

it'd be nice to avoid the `is_dummy` kwarg if at all possible. trying to keep the constructor interface lean n clean

Reply:
If we create the dummy at module import time, then it shouldn't be a problem. It's just that if I create the dummy for the first time upon the registration of a singleton int, or upon the construction of the first NestedTensor, then it would lead to infinite recursion. I will see if I can get rid if `is_dummy`.

Reply:
> If we create the dummy at module import time, then it shouldn't be a problem.

I take that back. The reason its tough is because we're doing - 
create dummy -> create nested tensor -> tries to register a symint for the dummy's offsets -> needs a dummy to register symint

Reply:
> trying to keep the constructor interface lean n clean

Technically its private though, so I'm kind of okay with it lol